common.js

```diff
@@ -49,3 +49,47 @@ function if_auto_decrypt(cb) {
         }
     });
 }
+
+// See https://gist.github.com/samgiles/762ee337dff48623e729
+// [B](f: (A) ⇒ [B]): [B]  ; Although the types in the arrays aren't strict (:
+Array.prototype.flatMap = function(lambda) {
```

Comment:
might want to call this `avd_flatmap` or somesuch so that if Github provide their own flatmap implementation later, we don't clobber it (which no doubt would be tough to debug)